frontend/src/search/SearchResult.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+import { Link, useSearchParams } from "react-router-dom";
+import { getUsers } from "../user/user-service";
+import User from "../user/user";
+import { useGlobalContext } from "../auth/GlobalContext";
+import { format } from "date-fns";
+
+const SearchResult: React.FC = () => {
+  const [searchParams] = useSearchParams();
+  const { refresh } = useGlobalContext();
+  const [results, setResults] = useState<User[]>([]);
+
+  useEffect(() => {
+    const resultHandler = async () => {
+      const query = searchParams.get("q");
+      if (!query) return;
+
+      const res = await getUsers();
+      setResults(res.filter((User) => User.displayName.includes?.(query)));
+    };
+    resultHandler();
+  }, [refresh]);
+
+  return (
+    <div className="flex flex-col gap-2 mt-2 mb-4">
+      {results.map((user) => (
+        <div
+          key={user.username}
+          className="my-2 py-2 pl-4 pr-4 glass rounded-lg text-left text-light gap-2 flex "
```

Comment:
It's preferable not to use margins inside a flex container if you can use gap instead. Remove the `my-2` and increase the parent div `gap-2`. If you want a margin on the first or last item, add padding to the parent div.